numpy/core/tests/test_multiarray.py

```diff
@@ -5637,24 +5637,27 @@ def test_native_padding_2(self):
         self._check('^x3T{xi}', {'f0': (({'f0': ('i', 1)}, (3,)), 1)})
 
     def test_trailing_padding(self):
-        # Trailing padding should be included, *and*, the item size
+        # Modified in 1.12. See gh-7797
+        # Trailing padding should cause the itemsize to increase,
+        # and makes non-structured datatypes into a structured
+        # type with a single field.
         # should match the alignment if in aligned mode
-        align = np.dtype('i').alignment
 
-        def VV(n):
-            return 'V%d' % (align*(1 + (n-1)//align))
+        def dtype_itemsize(n):
+            return {'names': ['f0'], 'formats': ['i'], 'itemsize': n}
 
-        self._check('ix', [('f0', 'i'), ('', VV(1))])
-        self._check('ixx', [('f0', 'i'), ('', VV(2))])
-        self._check('ixxx', [('f0', 'i'), ('', VV(3))])
-        self._check('ixxxx', [('f0', 'i'), ('', VV(4))])
-        self._check('i7x', [('f0', 'i'), ('', VV(7))])
+        # XXX not sure if this aligment is desired behavior
+        self._check('ix',    dtype_itemsize(8))
```

Comment:
The alignment in the native mode is supposed to follow that of the C compiler, so the test should check the results are in agreement with `dtype.alignment`